### PR TITLE
Move S.D.DiagnosticSource to next version

### PR DESCRIPTION
**Pull request by weshaggard**
PTAL @ericstj @lmolkova 

FYI @Petermarcu

### CODE REVIEW

File: src/System.Diagnostics.DiagnosticSource/pkg/System.Diagnostics.DiagnosticSource.pkgproj [hunk 1]
```diff
@@ -4,8 +4,6 @@
   <PropertyGroup>
     <!-- we need to be supported on pre-nuget-3 platforms (Dev12, Dev11, etc) -->
     <MinClientVersion>2.8.6</MinClientVersion>
-    <!-- workaround for https://github.com/dotnet/buildtools/issues/1529 -->
-    <HarvestVersion>4.3.0</HarvestVersion>
```
Comment (ericstj) [inline src/System.Diagnostics.DiagnosticSource/pkg/System.Diagnostics.DiagnosticSource.pkgproj]: Can you confirm that the packageIndex has the 4.4.0 version and that is what we are harvesting from during build?

**Comment by Petermarcu** [comment]
Go ahead and merge once we have core review signoff and green CI.

File: src/System.Diagnostics.DiagnosticSource/pkg/System.Diagnostics.DiagnosticSource.pkgproj (hunk 1 above)
Comment (weshaggard) [inline src/System.Diagnostics.DiagnosticSource/pkg/System.Diagnostics.DiagnosticSource.pkgproj]: I just pushed an update to add the 4.4.0 package restore in harvesting and updated the packageindex accordingly.

**Comment by weshaggard** [comment]
>Go ahead and merge once we have core review signoff and green CI.

I also want to wait until we get verification that the stable package works and is correct before merging this.

**Comment by ericstj** [comment]
You might want to add this to dir.props.
```
    <!-- DO NOT ship this as stable for .NET Core 2.0. -->
    <BlockStable>true</BlockStable>
```

We don't need to ship a servicing release of this when we ship netcoreapp2.0, so you can set this property as a bit of redundancy to prevent that.

**Comment by weshaggard** [comment]
Good idea @ericstj. Done. 

4.4.0 stable package is uploaded https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/4.4.0 so merging once CI shows green.

**Pull request merged**